sound/soc/sof/ipc4-topology.h

```diff
@@ -357,4 +357,35 @@ struct sof_ipc4_process {
 	bool payload_with_output_fmt;
 };
 
+/**
+ * struct sof_ipc4_pin_format - widgets pin format
+ * @pin_index: pin index for the widget
+ * @iobs: input/output buffer size for input/output pin
+ * @audio_fmt: audio format for the pin
+ */
+struct sof_ipc4_pin_format {
+	u32 pin_index;
+	u32 iobs;
+	struct sof_ipc4_audio_format audio_fmt;
+};
+
+/**
+ * struct sof_ipc4_base_module_cfg_ext - base module config extension
+ * @num_sink_pin_fmts: number of sink pin formats in the data member
+ * @num_source_pin_fmts: number of source pin formats in the data member
+ * @reserved: reserved for future use
+ * @priv_param_len: length of optional module specific parameters in the data member
+ * @data: consist of three parts:
+ *        a) num_sink_pin_fmts * sof_ipc4_pin_format followed by
+ *        b) num_source_pin_fmts * sof_ipc4_pin_format, and then followed by
```

Comment:
can a module have zero num_sink_pins && zero num_source_pins, and if yes does this impact how the data stuff is handled.

Reply:
we don't have process module that has zero num_sink_pins or zero num_source_pins, it is very unlikely we will have in the future. 
let me add comments for num_sink_pin_fmts and num_sink_pin_fmts to indicate that they are greater than zero.

Reply:
a comment alone won't be enough. You need the sanity check in the code when you use this